### PR TITLE
Add webpack 2.2.0 support.

### PR DESCRIPTION
**Pull request by jdalton**
This PR fixes compatibility with webpack 2.2.0.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -7,40 +7,31 @@ export default function (source, inputSourceMap) {
...
-      delete rollupOptions.plugins
-      Object.assign(rollupConfig, rollupOptions)
-    }
-  }
+  const parsed = loaderUtils.parseQuery(this.query)
+  const plugins = this.options.rollup
+  const options = plugins ? { plugins } : parsed
+
+  Object.assign(rollupConfig, options)
+  rollupConfig.plugins = [memory(), ...options.plugins || []]
```
Comment (jdalton) [inline src/index.js:26]: \cc @TheLarkInn 
Is this :banana:s or an 👌  way to go about supporting these [two config flavors](https://github.com/egoist/rollup-loader#usage)?

Comment (TheLarkInn) [reply src/index.js:26]: That looks okay to me.

**Comment by mattkime** [comment]
this may explain the disappointing results i saw w/ webpack 2.2

**Comment by egoist** [comment]
I guess `this.options.rollup` can be a plain object, not just an array of plugins?

**Comment by egoist** [comment]
maybe I missed something, what compatibility this PR fixes?

**Comment by NekR** [comment]
Yeah, why rollup options should be only plugins? What's wrong with current code/setup? What exactly this PR fixes, i.e. how it improves compatibility with webpack 2.2.0?

**Comment by jdalton** [comment]
@egoist 
> maybe I missed something, what compatibility this PR fixes?

It caused an error because `this.options.rollup` is `undefined` when running in Webpack 2.2.0. 
So then it would be added to the `rollupConfig.plugins` as `undefined` via `concat` and then rollup would error because of an `undefined` plugin being specified.

@NekR 
> Yeah, why rollup options should be only plugins? 

It's more than just `plugins`. In the case of Webpack 1 you can specify options as an array, to be just plugins, or a full options object. Webpack 2 has its own options format. There is just special handling of `plugins` to ensure the `rollup-plugin-memory` is loaded.

A bonus side effect of this PR is that loader queries are supported for WP1 too.

**Pull request merged**